include/boost/math/tools/polynomial.hpp

```diff
@@ -98,8 +111,125 @@ T evaluate_chebyshev(const Seq& a, const T& x)
    return a[0] / 2 + yk * x - yk1;
 }
 
+
+template <typename T>
+class polynomial;
+
+namespace detail {
+    
+/** 
+* Knuth, The Art of Computer Programming: Volume 2, Third edition, 1998
+* Chapter 4.6.1, Algorithm D: Division of polynomials over a field.
+* 
+* @tparam  T   Coefficient type, must be not be an integer.
+* 
+* Template-parameter T actually must be a field but we don't currently have that 
+* subtlety of distinction.
+*/
+template <typename T, typename N>
+BOOST_DEDUCED_TYPENAME disable_if_c<std::numeric_limits<T>::is_integer, void >::type
+division_impl(polynomial<T> &q, polynomial<T> &u, const polynomial<T>& v, N n, N k)
+{
+    q[k] = u[n + k] / v[n];
+    for (std::size_t j = n + k; j > k;)
+    {
+        j--;
+        u[j] -= q[k] * v[j - k];
+    }
+}
+
+
+/** 
+* Knuth, The Art of Computer Programming: Volume 2, Third edition, 1998
+* Chapter 4.6.1, Algorithm R: Pseudo-division of polynomials.
+* 
+* @tparam  T   Coefficient type, must be an integer.
+* 
+* Template-parameter T actually must be a unique factorization domain but we
+* don't currently have that subtlety of distinction.
+*/
+template <typename T, typename N>
+BOOST_DEDUCED_TYPENAME enable_if_c<std::numeric_limits<T>::is_integer, void >::type
+division_impl(polynomial<T> &q, polynomial<T> &u, const polynomial<T>& v, N n, N k)
+{
+    q[k] = u[n + k] * std::pow(v[n], k);
```

Comment:
> I might find a better algorithm for integer power, as I assume this is 
> not ideal.
>

Actually std::pow is generally pretty well optimized for that case - the 
"obvious" divide and conquer algorithm is often not much (or indeed any) 
faster.